src/consolidateListNodes.js

```diff
@@ -35,12 +36,144 @@ export default function consolidateListNodes(tr: Transform): Transform {
       tr = tr.insert(insertAt, content);
       prevJointInfo = jointInfo;
     } else {
+      tr = linkOrderedListCounters(tr);
       break;
     }
   }
   return tr;
 }
 
+/**
+ * This ensures that the adjacent ordered lists within the same indent level
+ * share the same counter.
+ *
+ * For example, the following three lists:
+ *   --------
+ *   1. AAA
+ *   2. BBB
+ *   --------
+ *     a. CCC
+ *     d. DDD
+ *   --------
+ *   1. EEE
+ *   2. FFF
+ *   --------
+ * Will transform into
+ *   --------
+ *   1. AAA
+ *   2. BBB
+ *   --------
+ *     a. CCC
+ *     d. DDD
+ *   --------
+ *   3. EEE
```

Comment:
what if the user wants this list to reset to 1?

Reply:
User need to explicitly hit an ENTER to create new paragraph before "3." and that will reset the number of this list.